packages/ui/src/components/Checkbox/SCheckbox.vue

```diff
@@ -0,0 +1,270 @@
+  }
+});
+</script>
+
+<template>
+  <label
+    :class="[
+      's-checkbox',
+      `s-checkbox--size-${size}`,
+      {
+        's-checkbox--bordered': border,
+        's-checkbox--checked': checked,
+        's-checkbox--disabled': disabled,
+        's-checkbox--indeterminate': indeterminate,
+      }
+    ]"
+  >
+    <div class="s-checkbox__content">
+      <span class="s-checkbox__input">
+        <input
+          :id="id"
+          :name="name"
+          :true-value="trueValue"
+          :false-value="falseValue"
+          :value="value"
+          :checked="checked"
+          :disabled="disabled"
+          :indeterminate="indeterminate"
+          v-model="model"
+          type="checkbox"
+          class="s-checkbox__original"
+        />
+        <span class="s-checkbox__inner" />
+      </span>
+      <span class="s-checkbox__label" v-if="$slots.default || label">
+        <slot>{{ label }}</slot>
+      </span>
+    </div>
+    <div class="s-checkbox__description" v-if="$slots.description">
+      <slot name="description" />
+    </div>
+  </label>
+</template>
+
+<style lang="scss">
+@mixin size($size, $padding, $labelOffset, $descriptionOffset) {
+  $base: calc(var(--s-size-#{$size}) / 2);
+
+  &.s-checkbox--size-#{$size} {
+    &.s-checkbox--bordered {
+      padding: $padding;
+    }
+
+    .s-checkbox__inner {
+      width: $base;
+      height: $base;
+    }
+
+    .s-checkbox__input + .s-checkbox__label {
+      margin-left: $labelOffset;
+    }
+
+    .s-checkbox__description {
+      margin-top: $descriptionOffset;
+    }
+  }
+}
+
+@mixin label-size($size, $fontSize, $lineHeight) {
+  &.s-checkbox--size-#{$size} {
+    .s-checkbox__label {
+      font-size: $fontSize;
+      line-height: $lineHeight;
+      letter-spacing: -0.02em;
+    }
+  }
+}
+
+@mixin checkmark-size($size, $top, $left, $width, $height) {
+  &.s-checkbox--size-#{$size} {
+    .s-checkbox__inner {
+      &::after {
+        top: $top;
+        left: $left;
+        width: $width;
+        height: $height;
+      }
+    }
+  }
+}
+
+.s-checkbox {
+  cursor: pointer;
+  display: inline-flex;
+  flex-flow: column nowrap;
+
+  @include size('mini', 7px 7px, 6px, 4px);
+  @include size('small', 7px 9px, 6px, 6px);
+  @include size('medium', 9px 9px, 8px, 8px);
+  @include label-size('mini', var(--s-font-size-2xs), 16px);
+  @include label-size('small', var(--s-font-size-xs), 18px);
+  @include label-size('medium', var(--s-font-size-sm), 20px);
+  @include checkmark-size('mini', 1px, 3px, 2px, 5px);
+  @include checkmark-size('small', 0px, 4px, 4px, 8px);
+  @include checkmark-size('medium', 1px, 6px, 5px, 10px);
+
+  &--indeterminate {
+    @include checkmark-size('mini', 4px, 2px, 6px, 2px);
+    @include checkmark-size('small', 6px, 3px, 8px, 2px);
+    @include checkmark-size('medium', 8px, 4px, 10px, 2px);
+  }
+
+  &--disabled {
+    cursor: not-allowed;
+    color: var(--s-color-base-on-disabled);
```

Comment:
Please never hardcode variables names - use JS bindings with `ThemeVars` or `@apply` windicss directive. Example:

```scss
&--disabled {
  @apply text-base-on-disabled;
}
```

Install Windi extension in VSCode for utility hints - it helps.

Also maybe it will be useful to expose some SCSS utility from `@soramitsu-ui/theme` package that will take to itself the responsibility of exact knowledge of variables names.

Why not use vars directly? To make code resistence to refactoring, thus more easy to support.